src/pages/søknad/steg/bo-og-opphold-i-norge/scheme.ts

```diff
@@ -0,0 +1,136 @@
+import { SøknadState } from '~src/features/søknad/søknad.slice';
+import { DelerBoligMed, EPSFormData } from '~src/features/søknad/types';
+import { keyOf } from '~src/lib/types';
+import yup from '~src/lib/validering';
+
+export type FormData = SøknadState['boOgOpphold'];
+
+const epsFormDataSchema = yup
+    .object<EPSFormData>({
+        fnr: yup.string().nullable().defined().length(11).typeError('Ugyldig fødselsnummer'),
+        alder: yup.number().nullable().defined(),
+        erUførFlyktning: yup
+            .boolean()
+            .when('alder', {
+                is: (val) => val < 67,
+                then: yup.boolean().required('Fyll ut om ektefelle/samboer er ufør flyktning'),
+                otherwise: yup.boolean().nullable().defined(),
+            })
+            .defined(),
+        eps: yup.mixed<Person>().test({
+            name: 'skal ha person dersom fnr er utfylt',
+            message: 'Ektefelle / samboer må være hentet av systemet, og tilgjengelig, før du kan fortsette',
+            test: function (val) {
+                const fnr: string = this.parent.fnr;
+
+                if (fnr?.length === 11) {
+                    return val !== null;
+                }
+                return false;
+            },
+        }),
+    })
+    .defined();
+
+export const schema = yup.object<FormData>({
+    borOgOppholderSegINorge: yup.boolean().nullable().required('Fyll ut om du bor og oppholder deg i Norge'),
+    delerBoligMedPersonOver18: yup.boolean().nullable().required('Fyll ut om du deler bolig med noen over 18 år'),
+    delerBoligMed: yup
+        .mixed<DelerBoligMed>()
+        .nullable()
+        .when('delerBoligMedPersonOver18', {
+            is: true,
+            then: yup
+                .mixed<DelerBoligMed>()
+                .nullable()
+                .oneOf<DelerBoligMed>(Object.values(DelerBoligMed), 'Du må velge hvem du deler bolig med')
+                .required(),
+        }),
+    ektefellePartnerSamboer: yup
+        .mixed<EPSFormData>()
+        .when(keyOf<FormData>('delerBoligMed'), {
+            is: DelerBoligMed.EKTEMAKE_SAMBOER,
+            then: epsFormDataSchema,
+            otherwise: yup.mixed().nullable().defined(),
+        })
+        .nullable()
+        .defined(),
+    innlagtPåInstitusjon: yup
+        .boolean()
+        .required('Fyll ut om du har vært innlagt på instituasjon siste 3 måneder')
+        .nullable(),
+    datoForInnleggelse: yup
+        .string()
+        .nullable()
+        .defined()
+        .when('innlagtPåInstitusjon', {
+            is: true,
+            then: yup
+                .string()
+                .required('Fyll ut datoen du var innlagt')
+                .test({
+                    name: 'datoFormattering',
+                    message: 'Dato må være formatert på dd.mm.yyyy',
+                    test: function (val) {
+                        return DateFns.isValid(DateFns.parse(val, 'yyyy-MM-dd', new Date()));
```

Comment:
`message` sier att det skal være dd.mm.yyyy men så parser vi etter yyyy-MM-dd. 🤔

Reply:
jeg er 90% på at dette er pga hvordan datepickeren lager datoer, og hvordan vi har lyst til å skrive datoer, og må gjøre noe triksing for at validering skal være OK 🤷 🤔

Reply:
Haha, ja den der er helt fucka. Vi viser den som dd.mm.yyyy, men den blir lagret som yyyy-mm-dd i staten vår (og dermed til db også). Tipper dette er et resultat av at backend sender (eller vil motta) som yyyy-mm-dd, men at det ikke er gir mening å vise det formatet til brukeren. Gadd uansett ikke løse opp det der i _denne_ pren, så lot det bare være som det var